openshift-provision/openshift-cluster-create.yaml

```diff
@@ -0,0 +1,119 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: openshift-install
+spec:
+  workspaces:
+    - name: install-dir
+      mountPath: /tekton/home/
+    - name: secrets
+  params:
+    - name: PLATFORM
+      type: string
+      description: Platform name where the openshift cluster needs to be deployed such as AWS, GCP etc
+      default: ""
+    - name: PROJECT_ID
+      type: string
+      default: ""
+      description: Project Id required in case of gcp
+    - name: CLUSTER_NAME
+      type: string
+      description: Cluster name
+    - name: BASE_DOMAIN
+      type: string
+      description: Base domain of your cluster install
+    - name: REPLICAS
+      description: How many replicas for the master and worker node
+      type: string
+    - name: REGION
+      type: string
+      description: AWS Region where to install
+  steps:
+    - name: openshift-install-config
+      image: registry.access.redhat.com/ubi8/ubi-minimal:latest
+      workingDir: $(workspaces.install-dir.path)
+      script: |
+        #!/usr/bin/env bash
+
+        echo "1 $(ls -a)"
+        credsPath=""
+        projectId=""
+        if [ "$(params.PLATFORM)" == "gcp" ]; then
+          credsPath+=".gcp"
+          mkdir .gcp
+          projectId+="ProjectID: $(params.PROJECT_ID)"
+        else
+          credsPath+=".aws"
+          mkdir .aws
+        fi
+
+        cp $(workspaces.secrets.path)/* $credsPath
+
+        echo "2 $(ls -a $credsPath/)"
+
+        cat $credsPath/credentials
+
+        [[ -f install-config.yaml ]] && \
+        echo 'using existing install-config.yaml' && \
+        cat install-config.yaml && exit 0
+
+        cat << EOF > install-config.yaml
+        apiVersion: v1
+        baseDomain: $(params.BASE_DOMAIN)
+        compute:
+        - hyperthreading: Enabled
+          name: worker
+          platform: {}
+          replicas: $(params.REPLICAS)
+        controlPlane:
+          hyperthreading: Enabled
+          name: master
+          platform: {}
+          replicas: $(params.REPLICAS)
+        metadata:
+          creationTimestamp: null
+          name: $(params.CLUSTER_NAME)
+        networking:
+          clusterNetwork:
+          - cidr: 10.128.0.0/14
+            hostPrefix: 23
+          machineCIDR: 10.0.0.0/16
+          networkType: OpenShiftSDN
+          serviceNetwork:
+          - 172.30.0.0/16
+        platform:
+          $(params.PLATFORM):
+            region: $(params.REGION)
+            ###ProjectID
```

Comment:
What is this projectid comment? why don't use just use $projectid directly instead of doing the sed?

Reply:
ProjectId is for the GCP support as `ProjectID: $(projectId)`
The ProjectID is specifically for gcp and not for aws

Reply:
`projectId+="ProjectID: $(params.PROJECT_ID)"` I am generating projectId like this

Reply:
https://github.com/tektoncd/catalog/pull/292/files/c585fc338bcba66446c124b7c557b329b4d38d6f#r415972600

Reply:
ok got it :sweat_smile: